src/transformers/models/depth_anything/modeling_depth_anything.py

```diff
@@ -0,0 +1,471 @@
+        """
+        Args:
+            hidden_states (`List[torch.FloatTensor]`, each of shape `(batch_size, sequence_length + 1, hidden_size)`):
+                List of hidden states from the backbone.
+        """
+        out = []
+
+        for i, hidden_state in enumerate(hidden_states):
+            # reshape to (batch_size, num_channels, height, width)
+            hidden_state = hidden_state[:, 1:]
+            batch_size, _, num_channels = hidden_state.shape
+            hidden_state = hidden_state.reshape(batch_size, patch_height, patch_width, num_channels)
+            hidden_state = hidden_state.permute(0, 3, 1, 2).contiguous()
+            hidden_state = self.layers[i](hidden_state)
+            out.append(hidden_state)
+
+        return out
+
+
+class DepthAnythingReassembleLayer(nn.Module):
+    def __init__(self, config, channels, factor):
+        super().__init__()
+        # projection
+        hidden_size = config.backbone_config.hidden_size
+        self.projection = nn.Conv2d(in_channels=hidden_size, out_channels=channels, kernel_size=1)
+
+        # up/down sampling depending on factor
+        if factor > 1:
+            self.resize = nn.ConvTranspose2d(channels, channels, kernel_size=factor, stride=factor, padding=0)
+        elif factor == 1:
+            self.resize = nn.Identity()
+        elif factor < 1:
+            # so should downsample
+            self.resize = nn.Conv2d(channels, channels, kernel_size=3, stride=int(1 / factor), padding=1)
+
+    # Copied from transformers.models.dpt.modeling_dpt.DPTReassembleLayer.forward
+    def forward(self, hidden_state):
+        hidden_state = self.projection(hidden_state)
+        hidden_state = self.resize(hidden_state)
+        return hidden_state
+
+
+class DepthAnythingFeatureFusionStage(nn.Module):
+    # Copied from transformers.models.dpt.modeling_dpt.DPTFeatureFusionStage.__init__ with DPT->DepthAnything
+    def __init__(self, config):
+        super().__init__()
+        self.layers = nn.ModuleList()
+        for _ in range(len(config.neck_hidden_sizes)):
+            self.layers.append(DepthAnythingFeatureFusionLayer(config))
+
+    def forward(self, hidden_states, size=None):
+        # reversing the hidden_states, we start from the last
+        hidden_states = hidden_states[::-1]
+
+        fused_hidden_states = []
+        # first layer only uses the last hidden_state
+        size = hidden_states[1].shape[2:]
+        fused_hidden_state = self.layers[0](hidden_states[0], size=size)
+        fused_hidden_states.append(fused_hidden_state)
+        # looping from the last layer to the second
+        for idx, (hidden_state, layer) in enumerate(zip(hidden_states[1:], self.layers[1:])):
+            if idx != len(hidden_states[1:]) - 1:
+                size = hidden_states[1:][idx + 1].shape[2:]
+            fused_hidden_state = layer(fused_hidden_state, hidden_state, size=size)
+            fused_hidden_states.append(fused_hidden_state)
```

Comment:
This can be simplified without having to use the conditional indexing logic 

```suggestion
        # We go backwards from the last hidden state to the first
        hidden_states = hidden_states[::-1]

        output_sizes = [hidden_state.shape[2:] for hidden_state in hidden_states[1:]]
        output_sizes += output_sizes[-1]

        # first layer only uses the last hidden_state
        residuals = [None] + hidden_states[1:]

        fused_hidden_states = []
        fused_hidden_state = hidden_states[0]
        
        # looping from the second last layer to the first
        for (residual, layer, output_size) in zip(residuals, self.layers, output_sizes):
            fused_hidden_state = layer(fused_hidden_state, residual=residual, size=output_size)
            fused_hidden_states.append(fused_hidden_state)
```

Reply:
This is not giving equivalent logits :/ there might be a bug in there